contracts/zero-ex/contracts/src/features/native_orders/NativeOrdersCancellation.sol

```diff
@@ -176,6 +177,38 @@ abstract contract NativeOrdersCancellation is
         }
     }
 
+    /// @dev Cancel all limit orders for a given maker and pair with a salt less
+    ///      than the value provided. The caller must a signer registered to the maker.
+    ///      Subsequent calls to this function with the same caller and pair require the
+    ///      new salt to be >= the old salt.
+    /// @param maker the maker for whom the msg.sender is the signer.
+    /// @param makerTokens The maker tokens.
+    /// @param takerTokens The taker tokens.
+    /// @param minValidSalts The new minimum valid salts.
+    function batchCancelPairLimitOrdersWithSigner(
+        address maker,
+        IERC20TokenV06[] memory makerTokens,
+        IERC20TokenV06[] memory takerTokens,
+        uint256[] memory minValidSalts
+    )
+        public
+    {
+        require(
+            makerTokens.length == takerTokens.length &&
+            makerTokens.length == minValidSalts.length,
+            "NativeOrdersFeature/MISMATCHED_PAIR_ORDERS_ARRAY_LENGTHS"
+        );
+
+        for (uint256 i = 0; i < makerTokens.length; ++i) {
+            cancelPairLimitOrdersWithSigner(
```

Comment:
might as well perform the signer check once and call `_cancelPairLimitOrders` here instead